### PR TITLE
Add CAN safety state led

### PR DESCRIPTION
**Pull request by CUAVcaijie**
Add CAN safety state led

### CODE REVIEW

File: src/drivers/uavcan/sensors/safety_state.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2014, 2015 PX4 Development Team. All rights reserved.
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/safety_state.cpp:3]:
```suggestion
 *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```

File: src/drivers/uavcan/sensors/safety_state.hpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+* LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+* ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+****************************************************************************/
+
+/**
+ * @file safety_state.hpp
+ *
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/safety_state.hpp:36]: Can you please add a description to what this represents using the `@brief` annotation?

File: src/drivers/uavcan/sensors/safety_state.hpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
+/****************************************************************************
+*
+*   Copyright (c) 2014, 2015 PX4 Development Team. All rights reserved.
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/safety_state.hpp:3]:
```suggestion
*   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```

File: src/drivers/uavcan/sensors/safety_state.hpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+#include <uavcan/uavcan.hpp>
+#include <com/hex/equipment/indication/SafetyStateCommand.hpp>
+#include <uORB/Subscription.hpp>
+#include <uORB/topics/actuator_armed.h>
+
+/**
+ * @brief The UavcanUavcanSafetyState class
+ */
+
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/safety_state.hpp:50]:
```suggestion
```

File: src/drivers/uavcan/sensors/safety_state.hpp [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+class UavcanUavcanSafetyState
+{
+public:
+	UavcanUavcanSafetyState(uavcan::INode &node);
+
+	/*
+	* setup periodic updater
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/safety_state.hpp:57]:
```suggestion
	 * setup periodic updater
```

File: src/drivers/uavcan/sensors/safety_state.hpp [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	*/
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/safety_state.hpp:58]:
```suggestion
	 */
```

File: src/drivers/uavcan/sensors/safety_state.hpp [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+	int init();
+
+private:
+	/*
+	 * Max update rate to avoid exessive bus traffic
+	 */
+	static constexpr unsigned			MAX_RATE_HZ = 10;
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/safety_state.hpp:65]:
```suggestion
	static constexpr unsigned MAX_RATE_HZ = 10;
```

File: src/drivers/uavcan/sensors/safety_state.hpp [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+	void periodic_update(const uavcan::TimerEvent &);
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/safety_state.hpp:67]: Can you add a description comment here as well?

File: src/drivers/uavcan/sensors/safety_state.hpp [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+	typedef uavcan::MethodBinder<UavcanUavcanSafetyState *, void (UavcanUavcanSafetyState::*)(const uavcan::TimerEvent &)>
+	TimerCbBinder;
+
+	/*
+	 * libuavcan related things
+	 */
+	uavcan::Publisher<com::hex::equipment::indication::SafetyStateCommand> _safety_state_pub;
+
+	uavcan::TimerEventForwarder<TimerCbBinder>			_timer;
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/safety_state.hpp:77]:
```suggestion
	uavcan::TimerEventForwarder<TimerCbBinder> _timer;
```

File: src/drivers/uavcan/sensors/safety_state.hpp [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+	 */
+	static constexpr unsigned			MAX_RATE_HZ = 10;
+
+	void periodic_update(const uavcan::TimerEvent &);
+
+	typedef uavcan::MethodBinder<UavcanUavcanSafetyState *, void (UavcanUavcanSafetyState::*)(const uavcan::TimerEvent &)>
+	TimerCbBinder;
+
+	/*
+	 * libuavcan related things
```
Comment (TSC21) [inline src/drivers/uavcan/sensors/safety_state.hpp:73]: Can you replace this with what exactly is being done here? Like initialising a UAVCAN `SafetyStateCommand` Publisher, create a method binder between a timer and the safety sate callback, etc.

File: src/drivers/uavcan/sensors/safety_state.hpp (hunk 2 above)
Comment (CUAVcaijie) [reply src/drivers/uavcan/sensors/safety_state.hpp:36]: ok,I will add

File: src/drivers/uavcan/sensors/safety_state.hpp (hunk 8 above)
Comment (CUAVcaijie) [reply src/drivers/uavcan/sensors/safety_state.hpp:67]: ok

File: src/drivers/uavcan/sensors/safety_state.hpp (hunk 10 above)
Comment (CUAVcaijie) [reply src/drivers/uavcan/sensors/safety_state.hpp:73]: Hello, are there any reference examples

**Pull request closed without merging**